### PR TITLE
Indicate whether an instruction is a "string" instruction

### LINKED ISSUE #184: RFC: Support for an `is_string_op` for LODS/STOS/MOVS/CMPS/SCAS
This family of instructions has interesting microarchitectural behavior (complex rollback handling, fast-string mode, etc.), and it'd be nice to be able to check whether a particular `Instruction` or `Code` is a string operation without having to compare against every member of the mnemonic/code family.

I'm happy to implement this; just wanted to check whether it's wanted first and whether you'd prefer it on `Instruction`, `Code`, or `Mnemonic` 🙂

### PR DESCRIPTION
**Pull request by woodruffw**
This is my first stab at #184 -- I ended up going the code/table generation route, since I initially missed your comment about doing this in the other `impl` block 😅 . Let me know if you'd prefer that I switch to what you said earlier, and I'll do so.

Closes #184.

### CODE REVIEW

File: src/rust/iced-x86/src/code.rs [hunk 1]
```diff
@@ -41705,4 +41705,43 @@ impl Code {
 			_ => false,
 		}
 	}
+
+	#[must_use]
+	#[inline]
+	pub(crate) fn is_string_op(self) -> bool {
```
Comment (wtfsck) [inline src/rust/iced-x86/src/code.rs:41711]: This should be public (and it needs docs). You need to add this attribute too `#[cfg(feature = "instr_info")]`

**Review by wtfsck — COMMENTED** [review]
Nice, and you even updated the C# code. :) Just one little thing then it's ready to merge!

**Comment by wtfsck** [comment]
> I ended up going the code/table generation route, since I initially missed your comment about doing this in the other `impl` block sweat_smile . Let me know if you'd prefer that I switch to what you said earlier, and I'll do so.

No this is good, it's similar to how I would've done it!

**Comment by wtfsck** [comment]
There's already an `impl` block with the attribute, so you could also move the fn there instead, see `condition_code()` instead.

**Comment by woodruffw** [comment]
> There's already an impl block with the attribute, so you could also move the fn there instead, see condition_code().

Will do!

**Comment by woodruffw** [comment]
Let me know if you'd like me to update the Python and JS bindings as well -- more than happy to 🙂

**Comment by wtfsck** [comment]
Yes, you can do that too. Thanks!

**Comment by woodruffw** [comment]
Done!

**Review by wtfsck — APPROVED** [review]
Great! Will merge it when the CI finishes successfully!

**Comment by woodruffw** [comment]
Awesome, thanks for the fast review and merge 😄

**Comment by wtfsck** [comment]
Ok, merged now, thanks!

**Pull request merged**